provider/clickhouse.go

```diff
@@ -954,6 +955,56 @@ func (store *clickHouseOfflineStore) CreateTrainingSet(def TrainingSetDef) error
 	return nil
 }
 
+func (store *clickHouseOfflineStore) CreateTrainingTestSplit(
+	trainingSetTable string,
+	testSize float32,
+	shuffle bool,
+	randomState int,
+) (string, error) {
+	// Generate unique suffix for the view names
+	tableNameSuffix := fmt.Sprintf("%s_%d_%t_%d", trainingSetTable, int(testSize*100), shuffle, randomState)
+	trainingSetSplitViewName := fmt.Sprintf("%s_split", tableNameSuffix)
+
+	// Generate ORDER BY clause for shuffling if needed
+	orderByClause := ""
+	if shuffle {
+		if randomState == 0 {
+			randomState = rand.Int() // Ensure a random state if 0 is provided
+		}
+		// Use ClickHouse's cityHash64 for deterministic shuffling, rowNumberInAllBlocks is a unique identifier for each row
+		orderByClause = fmt.Sprintf("ORDER BY cityHash64(concat(toString(rowNumberInAllBlocks()), toString(%d)))", randomState)
+	}
+
+	// Calculate the number of test rows based on the testSize parameter
+	var totalRows int
+	err := store.db.QueryRow(fmt.Sprintf("SELECT COUNT(*) FROM %s", sanitizeCH(trainingSetTable))).Scan(&totalRows)
+	if err != nil {
+		return "", fmt.Errorf("failed to get table size: %v", err)
+	}
+	testRows := int(float32(totalRows) * testSize)
+	fmt.Println("Number of test rows: ", testRows)
+
+	// Create the final view with an 'is_test' column
+	createViewQuery := fmt.Sprintf(`
+			CREATE VIEW IF NOT EXISTS %s AS
+			SELECT *, IF(row_number() OVER (%s) <= %d, 1, 0) AS is_test
+			FROM %s
+    	`,
+		sanitizeCH(trainingSetSplitViewName),
+		orderByClause,
+		testRows,
+		sanitizeCH(trainingSetTable),
+	)
+
+	fmt.Println("This is the create view query:", createViewQuery)
```

Comment:
should remove this

Reply:
these prints are actually extremely useful for logging